runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -555,6 +594,37 @@ private void onConfigChanged(ConfigChanged event)
 		{
 			removeCounter();
 		}
+
+		if (config.showBraceletInfobox())
+		{
+			clientThread.invoke(this::addBraceletCounter);
+		}
+		else
+		{
+			removeBraceletCounter();
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(ItemContainerChanged event)
+	{
+		if (event.getItemContainer() != client.getItemContainer(InventoryID.EQUIPMENT))
+		{
+			return;
+		}
+
+		Item[] equipment = event.getItemContainer().getItems();
+		int gloveSlotID = EquipmentInventorySlot.GLOVES.getSlotIdx();
+
+		if (equipment.length <= gloveSlotID)
+		{
+			return;
+		}
+
+		equippedSlaughter = equipment[gloveSlotID].getId() == ItemID.BRACELET_OF_SLAUGHTER;
+		equippedExpeditious = equipment[gloveSlotID].getId() == ItemID.EXPEDITIOUS_BRACELET;
+		removeBraceletCounter();
```

Comment:
just update the infobox instead of removing and readding it. it makes no sense to do it here

Reply:
The logic looks pretty ugly if I do it that way https://pastebin.com/DhQu6QVy

Also, #7620 and `setTask()` both handle it by removing and readding the infobox